src/extension/devtools/devtools.ts

```diff
@@ -38,71 +111,101 @@ function sendMessageToClient(message: string) {
 }
 
 function startRequestInterval(ms = 500) {
-  sendMessageToClient(REQUEST_DATA);
-  const id = setInterval(sendMessageToClient, ms, REQUEST_DATA);
+  let id: NodeJS.Timeout;
+
+  if (devtoolsMachine.matches("connected")) {
+    sendMessageToClient(REQUEST_DATA);
+    id = setInterval(sendMessageToClient, ms, REQUEST_DATA);
+  }
+
   return () => clearInterval(id);
 }
 
-let isPanelCreated = false;
-let isAppInitialized = false;
-
 devtools.addConnection(EXPLORER_SUBSCRIPTION_TERMINATION, () => {
   sendMessageToClient(EXPLORER_SUBSCRIPTION_TERMINATION);
 });
 
-devtools.listen<string>(CREATE_DEVTOOLS_PANEL, async ({ payload }) => {
-  if (isPanelCreated) {
-    return;
-  }
+const unsubscribers = new Set<() => void>();
+
+function unsubscribeFromAll() {
+  unsubscribers.forEach((unsubscribe) => unsubscribe());
+  unsubscribers.clear();
+}
 
+let connectedToPanel = false;
+
+async function createDevtoolsPanel() {
```

Comment:
This is the biggest change in the whole PR. Rather than waiting for a `CREATE_DEVTOOLS_PANEL` message, we unconditionally create it. See the PR description for why I believe this is a good idea 🙂